module2/gift-certificate-system/service/src/test/java/com/epam/esm/service/GiftCertificateServiceImplTest.java

```diff
@@ -0,0 +1,167 @@
+package com.epam.esm.service;
+
+import com.epam.esm.dao.GiftCertificateDao;
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dao.TagDaoJdbc;
+import com.epam.esm.exception.GiftCertificateNotFoundException;
+import com.epam.esm.model.GiftCertificate;
+import com.epam.esm.model.Tag;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import javax.sql.DataSource;
+import java.math.BigDecimal;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(locations = {"classpath:test-db.xml"})
+class GiftCertificateServiceImplTest {
+
+    private GiftCertificateService certificateService;
+
+    @Autowired
+    private DataSource dataSource;
```

Comment:
Mock dependency to DAO layer. You have already tested persistence layer in the corresponding module.